### ISSUE TITLE
Question: ports? / reverse proxy

### ISSUE THREAD

**Issue by PeterNerlich** [shaderecker/ansible-pihole#10]
Thank you for this project! It was immensely helpful getting started with Pi-hole quickly while having peace of mind that I won't have to spend time if I need to fix something later.

I want to add a service to my RPi that should be available as a custom domain, so I need to setup a reverse proxy. However, I noticed that I don't understand how this project achieves the Pi-hole docker container to be available from outside. I don't see the ports recommended in the README of docker-pi-hole reflected in the corresponding main.yml of roles/pihole/tasks (starting line 31). I tried to look around a bit but I'm lost.

Would you mind explaining to me where I need to point incoming connections to a reverse proxy instead of Pi-hole directly? I would probably either use traefik in docker or have a look at running nginx on the host, I'm not sure yet. Thanks!

**Comment by PeterNerlich** [comment]
On a related note, I tried changing the password for the web interface after having it left at the default initially, but after a run it was still the old one. I think I noticed previously that ansible wouldn't restart a container if a label changed but the container was still running, so I stopped the container and tried again. But the password still didn't change. If you want, I can open an issue for that

**Comment by PeterNerlich** [comment]
Oh, I got it! Because it uses the host network mode, it's not on a separated, virtual net and every port opened *is* the one of the host.

So I just need to change that, publish port 53 including udp (I don't know if a reverse proxy would handle that) and 80 via the reverse proxy. (Please correct me if you see an obvious error)

**Comment by shaderecker** [comment]
Hi @PeterNerlich yes exactly, currently it is configured to use host network mode.
And yes, if you want to change that you need to expose port 53 TCP & UDP, port 80 TCP, and optionally port 67 UDP if you are using the pihole DHCP server.

**Comment by shaderecker** [comment]
> On a related note, I tried changing the password for the web interface after having it left at the default initially, but after a run it was still the old one. I think I noticed previously that ansible wouldn't restart a container if a label changed but the container was still running, so I stopped the container and tried again. But the password still didn't change. If you want, I can open an issue for that

Hmm that's interesting. I think I never tried to change it afterwards in the past. It might be that the password is written into some pihole config file during the first container start and then afterwards not overwritten. I will dig into it.

**Comment by shaderecker** [comment]
After digging into this I figured out that the issue must be in the pihole container itself.
Ansible notices the password change and applies it to the container and also restarts the container.
I see the new password in the container env variables if I check with `docker inspect pihole`afterwards.
When checking the container logs with `docker logs pihole` I saw this message:
`::: Pre existing WEBPASSWORD found`
So it seems to find the existing `WEBPASSWORD` variable in `/home/pi/pihole/pihole/setupVars.conf` and uses that instead of applying the new password.
I will open a bug in the docker pihole repo.
